src/Analyzers/CSharp/Analyzers/UseAutoProperty/CSharpUseAutoPropertyAnalyzer.cs

```diff
@@ -95,33 +96,63 @@ SyntaxKind.RecordDeclaration or
                         if (addressOfExpression.Kind() == SyntaxKind.AddressOfExpression)
                             AddIneligibleFields(semanticModel, addressOfExpression.Operand, ineligibleFields, cancellationToken);
                     }
+
+                    foreach (var memberAccess in typeDeclaration.DescendantNodesAndSelf().OfType<MemberAccessExpressionSyntax>())
+                        AddIneligibleFieldsIfAccessedOffNotDefinitelyAssignedValue(semanticModel, memberAccess, ineligibleFields, cancellationToken);
```

Comment:
I worry about the placement of this. I would expect this to be efficient if we made it an IOperation-based symbol start/stop analyzer, where we track these ineligible fields as we come across them and not try and re-walk the type later.

Reply:
it's effectively equally as inefficient as our current approach for everything else in this type.  :)